src/codegen.cpp

```diff
@@ -1101,11 +1101,7 @@ jl_code_instance_t *jl_compile_linfo(jl_method_instance_t *mi, jl_code_info_t *s
                     if (codeinst->min_world <= world && world <= codeinst->max_world) {
                         jl_llvm_functions_t decls = codeinst->functionObjectsDecls;
                         bool already_compiled = params->cached && decls.functionObject != NULL;
-                        if (!src) {
-                            if (already_compiled || codeinst->invoke == jl_fptr_const_return)
-                                goto locked_out;
-                        }
-                        else if (already_compiled) {
+                        if (already_compiled) {
```

Comment:
@vtjnash if you could check this is correct.

Reply:
Since this preserves the current behavior (I'm just removing the dead code), I'll merge this, but would be good for you to look at to see if you meant something else.